[19:43:14.416][ INFO][Application][main]: starting service 'Socks5' on ip: 192.168.1.11and port: 12345
[19:43:14.417][ INFO][Application][main]: application is lunched.
[19:43:14.426][TRACE][Socks5][Thread-0]: bound to 192.168.1.11:12345
[19:43:14.429][TRACE][Socks5][Thread-0]: thread pool is built.
[19:44:14.519][ INFO][Socks5][Thread-0]: New client is joined from 192.168.1.11:63176
[19:44:14.524][ INFO][Socks5][Thread-0]: sent to be served.
[19:44:14.524][ INFO][Socks5][Thread-0]: New client is joined from 192.168.1.11:63177
[19:44:14.525][ INFO][Socks5][Thread-0]: sent to be served.
[19:44:14.525][ INFO][Socks5][Thread-0]: New client is joined from 192.168.1.11:63178
[19:44:14.525][ INFO][Socks5][Thread-0]: sent to be served.
[19:44:14.538][TRACE][MainService][pool-1-thread-2]: [][       2]methodFlow is passed!
[19:44:14.538][TRACE][MainService][pool-1-thread-1]: [][       1]methodFlow is passed!
[19:44:14.538][TRACE][MainService][pool-1-thread-3]: [][       3]methodFlow is passed!
[19:44:14.542][ERROR][MainService][pool-1-thread-2]: [][       2]Service error
java.lang.NullPointerException
	at azizollahi.app.socks.core.flow.CommandFlow.AnswerBack(CommandFlow.java:81)
	at azizollahi.app.socks.core.flow.CommandFlow.Process(CommandFlow.java:50)
	at azizollahi.app.socks.core.ClientService.run(ClientService.java:37)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.base/java.lang.Thread.run(Thread.java:844)
[19:44:14.542][ERROR][MainService][pool-1-thread-1]: [][       1]Service error
java.lang.NullPointerException
	at azizollahi.app.socks.core.flow.CommandFlow.AnswerBack(CommandFlow.java:81)
	at azizollahi.app.socks.core.flow.CommandFlow.Process(CommandFlow.java:50)
	at azizollahi.app.socks.core.ClientService.run(ClientService.java:37)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.base/java.lang.Thread.run(Thread.java:844)
[19:44:14.607][ INFO][Socks5][Thread-0]: New client is joined from 192.168.1.11:63179
[19:44:14.607][ INFO][Socks5][Thread-0]: sent to be served.
[19:44:14.608][TRACE][MainService][pool-1-thread-4]: [][       4]methodFlow is passed!
[19:44:14.726][ INFO][Socks5][Thread-0]: New client is joined from 192.168.1.11:63180
[19:44:14.727][ INFO][Socks5][Thread-0]: sent to be served.
[19:44:14.727][TRACE][MainService][pool-1-thread-5]: [][       5]methodFlow is passed!
[19:44:14.750][DEBUG][MainService][pool-1-thread-4]: [][       4]command flow is passed!
[19:44:14.750][DEBUG][MainService][pool-1-thread-3]: [][       3]command flow is passed!
[19:44:14.750][DEBUG][MainService][pool-1-thread-5]: [][       5]command flow is passed!
[19:44:14.752][ INFO][MainService][pool-1-thread-3]: [][       3]The channel bridge is started serving
[19:44:14.752][ INFO][MainService][pool-1-thread-5]: [][       5]The channel bridge is started serving
[19:44:14.752][ INFO][MainService][pool-1-thread-4]: [][       4]The channel bridge is started serving
[19:44:14.754][ INFO][MainService][Thread-3]: [][       5]Software caused connection abort: recv failed breaking bridge!
[19:44:14.754][ INFO][MainService][Thread-2]: [][       3]Software caused connection abort: recv failed breaking bridge!
[19:44:14.754][ INFO][MainService][Thread-1]: [][       4]Software caused connection abort: recv failed breaking bridge!
[19:44:14.754][ INFO][MainService][Thread-6]: [][       5]Socket closed breaking bridge!
[19:44:14.754][ INFO][MainService][Thread-5]: [][       4]Socket closed breaking bridge!
[19:44:14.754][ INFO][MainService][Thread-4]: [][       3]Socket closed breaking bridge!
[19:44:15.385][ INFO][Socks5][Thread-0]: New client is joined from 192.168.1.11:63184
[19:44:15.386][ INFO][Socks5][Thread-0]: sent to be served.
[19:44:15.386][TRACE][MainService][pool-1-thread-6]: [][       6]methodFlow is passed!
[19:44:15.404][ INFO][Socks5][Thread-0]: New client is joined from 192.168.1.11:63186
[19:44:15.404][ INFO][Socks5][Thread-0]: sent to be served.
[19:44:15.405][TRACE][MainService][pool-1-thread-7]: [][       7]methodFlow is passed!
[19:44:15.409][ INFO][Socks5][Thread-0]: New client is joined from 192.168.1.11:63187
[19:44:15.409][ INFO][Socks5][Thread-0]: sent to be served.
[19:44:15.410][TRACE][MainService][pool-1-thread-8]: [][       8]methodFlow is passed!
[19:44:15.411][DEBUG][MainService][pool-1-thread-8]: [][       8]command flow is passed!
[19:44:15.411][ INFO][MainService][pool-1-thread-8]: [][       8]The channel bridge is started serving
[19:44:15.591][DEBUG][MainService][pool-1-thread-7]: [][       7]command flow is passed!
[19:44:15.592][DEBUG][MainService][pool-1-thread-6]: [][       6]command flow is passed!
[19:44:15.592][ INFO][MainService][pool-1-thread-7]: [][       7]The channel bridge is started serving
[19:44:15.593][ INFO][MainService][pool-1-thread-6]: [][       6]The channel bridge is started serving
[19:44:16.065][ INFO][Socks5][Thread-0]: New client is joined from 192.168.1.11:63192
[19:44:16.065][ INFO][Socks5][Thread-0]: sent to be served.
[19:44:16.065][TRACE][MainService][pool-1-thread-9]: [][       9]methodFlow is passed!
[19:44:16.065][ERROR][MainService][pool-1-thread-9]: [][       9]Service error
java.lang.NullPointerException
	at azizollahi.app.socks.core.flow.CommandFlow.AnswerBack(CommandFlow.java:81)
	at azizollahi.app.socks.core.flow.CommandFlow.Process(CommandFlow.java:50)
	at azizollahi.app.socks.core.ClientService.run(ClientService.java:37)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:514)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.base/java.lang.Thread.run(Thread.java:844)
[19:44:16.466][ INFO][Socks5][Thread-0]: New client is joined from 192.168.1.11:63193
[19:44:16.467][ INFO][Socks5][Thread-0]: sent to be served.
[19:44:16.467][TRACE][MainService][pool-1-thread-10]: [][       a]methodFlow is passed!
[19:44:16.694][DEBUG][MainService][pool-1-thread-10]: [][       a]command flow is passed!
[19:44:16.695][ INFO][MainService][pool-1-thread-10]: [][       a]The channel bridge is started serving
[19:45:17.092][ INFO][MainService][Thread-11]: [][       6]peer socket is closed breaking bridge!
[19:45:17.092][ INFO][MainService][Thread-9]: [][       7]peer socket is closed breaking bridge!
[19:45:17.093][ INFO][MainService][Thread-12]: [][       6]Socket closed breaking bridge!
[19:45:17.093][ INFO][MainService][Thread-10]: [][       7]Socket closed breaking bridge!
[19:47:07.678][ INFO][MainService][Thread-13]: [][       a]peer socket is closed breaking bridge!
[19:47:07.679][ INFO][MainService][Thread-7]: [][       8]peer socket is closed breaking bridge!
[19:47:07.679][ INFO][MainService][Thread-14]: [][       a]Socket closed breaking bridge!
[19:47:07.680][ INFO][MainService][Thread-8]: [][       8]Socket closed breaking bridge!
